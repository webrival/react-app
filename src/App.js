import React, { Component } from 'react';
import './App.css';
import { BrowserRouter as Router, Route, Switch, Link } from 'react-router-dom';
import { Home, About, Contact, Navbar } from './component';
import Footer from './component/Footer';

class App extends Component {
  /**
   * Lifecycle Method
   */

  constructor(props) {
    super(props);

    console.log('Constructor');
  }

  componentDidMount() {
    console.log('Component did mount');
  }

  componentWillUnmount() {
    console.log('Component will unmount!');
  }

  render() {
    return (
      <div className='page-container'>
        <div className='content-wrap'>
          <Router>
            <Navbar />
            <Link to='/'></Link>
            <Link to='/about'></Link>

            <Switch>
              <Route exact path='/'>
                <Home />
              </Route>
              <Route path='/about'>
                <About />
              </Route>
              <Route path='/contact'>
                <Contact />
              </Route>
            </Switch>
          </Router>
        </div>
        <Footer />
      </div>
    );
  }
}

export default App;
