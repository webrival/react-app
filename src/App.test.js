import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';

test('Snapshot App Component', () => {
  const component = renderer.create(<App />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
