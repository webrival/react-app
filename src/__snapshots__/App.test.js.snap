// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot App Component 1`] = `
<div
  className="page-container"
>
  <div
    className="content-wrap"
  >
    <div>
      <nav
        className="navbar navbar-expand-lg navbar-dark bg-dark"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          Prototype
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            className="mr-auto navbar-nav"
            onKeyDown={[Function]}
          />
          <div
            className="navbar-nav"
            onKeyDown={[Function]}
          >
            <div
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </div>
            <div
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/about"
                onClick={[Function]}
              >
                About
              </a>
            </div>
            <div
              className="nav-item"
            >
              <a
                className="nav-link mr-sm-5"
                href="/contact"
                onClick={[Function]}
              >
                Contact Me
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
    <a
      href="/"
      onClick={[Function]}
    />
    <a
      href="/about"
      onClick={[Function]}
    />
    <div
      className="header"
    >
      <div
        className="company-pic"
      />
      <div
        className="company-text"
      >
        <h1>
          Welcome to Prototype.
        </h1>
        <p>
          an company that provide course about Web Development.
        </p>
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Read More
        </button>
      </div>
    </div>
    <div
      className="information"
    >
      <div
        className="information-text"
      >
        <p>
          Learn with Us, course that we provide for you
        </p>
        <ul>
          <li
            className="course-list"
          >
            <a
              className="button1"
              href="#"
            >
              HTML
            </a>
          </li>
          <li
            className="course-list"
          >
            <a
              className="button1"
              href="#"
            >
              CSS
            </a>
          </li>
          <li
            className="course-list"
          >
            <a
              className="button1"
              href="#"
            >
              Javascript
            </a>
          </li>
          <li
            className="course-list"
          >
            <a
              className="button1"
              href="#"
            >
              React
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="main-section"
    >
      <div
        className="content"
      >
        <h4>
          My Website
        </h4>
        <p>
          We spend so much of our life looking - but never seeing. In your world you can create anything you desire. Let's put some highlights on these little trees. The sun wouldn't forget them. Isn't that fantastic that you can create an almighty tree that fast? Just go back and put one little more happy tree in there. A thin paint will stick to a thick paint.
        </p>
        <p>
          You can't make a mistake. Anything that happens you can learn to use - and make something beautiful out of it. Volunteering your time; it pays you and your whole community fantastic dividends. We have a fantastic little sky! Maybe there's a little something happening right here. I thought today we would do a happy little picture. No worries. No cares. Just float and wait for the wind to blow you around.
        </p>
        <div
          className="content-pic"
        />
        <p>
          We don't have to be concerned about it. We just have to let it fall where it will. Don't be afraid to make these big decisions. Once you start, they sort of just make themselves. Let's put some happy little bushes on the other side now. Just think about these things in your mind and drop em' on canvas. It all happens automatically.
        </p>
        <p>
          Every single thing in the world has its own personality - and it is up to you to make friends with the little rascals. Everyone needs a friend. Friends are the most valuable things in the world. I thought today we would do a happy little picture. Trees live in your fan brush, but you have to scare them out. See how easy it is to create a little tree right in your world. We need a shadow side and a highlight side. Only eight colors that you need. We're not trying to teach you a thing to copy. We're just here to teach you a technique, then let you loose into the world.
        </p>
      </div>
    </div>
  </div>
  <div
    className="main-footer"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h4>
            Lorem Ipsum
          </h4>
          <ul
            className="list-unstyled"
          >
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
          </ul>
        </div>
        <div
          className="col"
        >
          <h4>
            Lorem Ipsum
          </h4>
          <ul
            className="list-unstyled"
          >
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
          </ul>
        </div>
        <div
          className="col"
        >
          <h4>
            Lorem Ipsum
          </h4>
          <ul
            className="list-unstyled"
          >
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
          </ul>
        </div>
        <div
          className="col"
        >
          <h4>
            Lorem Ipsum
          </h4>
          <ul
            className="list-unstyled"
          >
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
            <li>
              Lorem Ipsum
            </li>
          </ul>
        </div>
      </div>
      <div
        className="footer-bottom"
      >
        <p
          className="text-xs-center"
        >
          Â©
          2020
           Prototype.Inc | All right reserved | Terms Of Service | Privacy
        </p>
      </div>
    </div>
  </div>
</div>
`;
